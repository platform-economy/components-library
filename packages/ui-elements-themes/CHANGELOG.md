# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
