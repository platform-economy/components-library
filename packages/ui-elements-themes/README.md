# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
