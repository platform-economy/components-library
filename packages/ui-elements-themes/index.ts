export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
