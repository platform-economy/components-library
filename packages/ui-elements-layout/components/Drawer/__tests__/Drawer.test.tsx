import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';

import 'jest-styled-components';

import { Drawer, getDefaultUnit, getHeight, getWidth, DrawerProps } from '../index';

import { withTestTheme } from '@relayr/ui-elements-themes';

describe('ui-elements-layout', () => {
  describe('Components', () => {
    describe('Drawer', () => {
      it('Renders Drawer', () => {
        const tree = TestRenderer
          .create(withTestTheme(
          <Drawer size={10} open={true} direction="vertical" />,
          ))
          .toJSON();

        expect(tree).toMatchSnapshot();
      });

      it('Renders Drawer with children', () => {
        const tree = TestRenderer
          .create(withTestTheme(
            <Drawer size={10} open={true} direction="vertical" >
              Content
            </Drawer>,
          ))
          .toJSON();

        expect(tree).toMatchSnapshot();
      });

      it('Horizontal Drawer has overflow set to "hidden auto"', () => {
        const tree = TestRenderer.create(withTestTheme(
        <Drawer size={10} direction="horizontal" />,
        ))
          .toJSON();
        expect(tree).toHaveStyleRule('overflow', 'hidden auto');
      });

      it('Vertical Drawer has overflow set to "hidden hidden"', () => {
        const tree = TestRenderer.create(withTestTheme(
        <Drawer size={10} direction="vertical" />,
        ))
        .toJSON();
        expect(tree).toHaveStyleRule('overflow', 'hidden hidden');
      });

      it('getDefaultUnit returns string with appropriate unit', () => {
        expect(getDefaultUnit(1)).toBe('1px');
        expect(getDefaultUnit(2, 'em')).toBe('2em');
        expect(getDefaultUnit('3', 'em')).toBe('3');
      });

      it('getWidth returns 100% if direction is vertical', () => {
        const props = {
          direction: 'vertical',
          size: 10,
        } as DrawerProps;

        expect(getWidth(props)).toBe('100%');
      });

      it('getHeight returns 100% if direction is horizontal', () => {
        const props = {
          direction: 'horizontal',
          size: 10,
        } as DrawerProps;

        expect(getHeight(props)).toBe('100%');
      });

      it('getWidth returns 0px for horizontal direction and unset open prop', () => {
        const props = {
          direction: 'horizontal',
          size: 10,
        } as DrawerProps;

        expect(getWidth(props)).toBe('0px');
      });

      it('getHeight returns 0px for vertical direction and unset open prop', () => {
        const props = {
          direction: 'vertical',
          size: 10,
        } as DrawerProps;

        expect(getHeight(props)).toBe('0px');
      });

      it('getWidth returns size for horizontal direction and set open prop', () => {
        const props = {
          direction: 'horizontal',
          size: '10px',
          open: true,
        } as DrawerProps;

        expect(getWidth(props)).toBe('10px');
      });

      it('getHeight returns size for vertical direction and set open prop', () => {
        const props = {
          direction: 'vertical',
          size: '10px',
          open: true,
        } as DrawerProps;

        expect(getHeight(props)).toBe('10px');
      });
    });
  });
});
