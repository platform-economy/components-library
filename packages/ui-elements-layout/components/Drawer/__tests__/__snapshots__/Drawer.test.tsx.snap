// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-layout Components Drawer Renders Drawer 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
  height: 10px;
  -webkit-transition-property: width,height;
  transition-property: width,height;
  -webkit-transition-duration: .2s;
  transition-duration: .2s;
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  background-color: #surface;
  overflow: hidden hidden;
}

<div
  className="c0"
  direction="vertical"
  open={true}
  size={10}
/>
`;

exports[`ui-elements-layout Components Drawer Renders Drawer with children 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
  height: 10px;
  -webkit-transition-property: width,height;
  transition-property: width,height;
  -webkit-transition-duration: .2s;
  transition-duration: .2s;
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  background-color: #surface;
  overflow: hidden hidden;
}

<div
  className="c0"
  direction="vertical"
  open={true}
  size={10}
>
  Content
</div>
`;
